src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs

```diff
@@ -25,6 +25,38 @@ public sealed partial class AssemblyName : ICloneable, IDeserializationCallback,
         private AssemblyVersionCompatibility _versionCompatibility;
         private AssemblyNameFlags _flags;
 
+        public AssemblyName(string assemblyName)
+            : this()
+        {
+            if (assemblyName == null)
+                throw new ArgumentNullException(nameof(assemblyName));
+            if ((assemblyName.Length == 0) ||
+                (assemblyName[0] == '\0'))
+                throw new ArgumentException(SR.Format_StringZeroLength);
+
+            AssemblyNameParser.AssemblyNameParts parts = AssemblyNameParser.Parse(assemblyName);
+            this.Name = parts._name;
```

Comment:
Nit: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md: We avoid `this.` unless absolutely necessary.

Also, why not assign the fields directly?

Reply:
Some of these properties are not just assigning fields. I.E.  `CultureName` specialcases `null`, `Flags` does some masking.
I did not want to have a mix of field and property assignments.

Reply:
There is a lot of busy work happening by going through properties, in particular with decomposing and composing the flags again.

If you just went with assigning fields, the whole thing becomes just:

```
            _name = parts._name;
            _version = parts._version;
            _flags = parts._flags;
            _publicKeyOrToken = parts._publicKeyOrToken;
            if (parts._cultureName != null)
                _cultureInfo = new CultureInfo(parts._cultureName);
```